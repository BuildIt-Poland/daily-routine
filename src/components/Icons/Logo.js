import React from 'react';

function Logo() {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width="126"
      height="20"
      viewBox="0 0 126 20"
      role="img"
      aria-labelledby="icon-logo-title"
    >
      <title id="icon-logo-title">Daily Routine</title>
      <path
        fill="#ff1d12"
        d="M27.033 2.601c0-.68.52-1.18 1.24-1.18.68 0 1.2.4 1.2 1.16 0 .72-.46 1.24-1.24 1.24-.7 0-1.2-.46-1.2-1.22zm22.275 3.36v7.979c0 3.52-1.66 5.739-4.879 5.739-1.8 0-3.1-1.06-3.1-1.54 0-.3.72-1.28 1-1.28.08 0 .16.06.34.2.42.34 1.04.76 1.86.76 1.5 0 2.82-1 2.82-3.96l.02-2.019-.08-.02c-.52 1.4-1.74 2.8-3.599 2.8-1.17 0-2.018-.513-2.463-1.46-.936.887-2.235 1.46-3.888 1.46-1.678 0-2.978-.682-3.775-1.846-.748 1.014-1.849 1.846-3.351 1.846-1.196 0-2.099-.512-2.598-1.434-.714.799-1.694 1.434-2.963 1.434-1.4 0-2.4-.76-2.4-1.96v-.3l-.08-.02c-.48 1.14-1.6 2.28-3.44 2.28-1.971 0-3.434-1.437-3.738-3.648-.947 2.283-2.974 3.988-6.077 3.988-1.04 0-2.36-.16-3.5-.16-1.12 0-1.82.16-1.94.16-.14 0-.22-.08-.26-.26-.12-.36-.2-.96-.2-1.24 0-.16.1-.24.34-.28.58-.1 1.3-.14 1.84-.14h.24c.36-.48.44-1.3.44-2.92 0-1.96-.06-3.92-.06-5.439 0-.34.14-.5.48-.5h1.2c.24 0 .4.18.4.48V9.92c-.02 1.58-.1 2.639-.58 3.239.44.02.88.04 1.32.04 3.08 0 4.759-2.499 4.759-5.459 0-3.28-2.44-5.559-5.939-5.559-2.82 0-4.819 1.76-4.819 3.839 0 1.24.74 2.06 1.5 2.4.1.06.14.08.14.18 0 .22-.5 1.08-.68 1.32-.04.06-.1.12-.2.12-.48 0-2.74-1.12-2.74-4.019 0-3.399 3.06-5.699 6.839-5.699 4.652 0 7.938 2.899 8.111 7.073.738-1.305 2.027-2.134 3.604-2.134 1.26 0 2.2.5 2.84 1.08l.08-.04v-.36c0-.28.16-.44.4-.44h1.26c.26 0 .42.16.42.46 0 .38-.24 3.46-.24 5.039 0 1 .4 1.78 1.2 1.78.815 0 1.505-.756 2.002-1.604V5.961c0-.28.14-.46.42-.46h1.28c.3 0 .44.16.44.48 0 .92-.08 3.699-.08 4.739 0 1.24.44 2.06 1.44 2.06.91 0 1.667-.861 2.165-1.714a6.631 6.631 0 0 1-.159-1.465V7.1c0-3.659 1.36-6.779 4.299-6.779 1.78 0 2.9 1.22 2.9 2.86 0 2.42-2.44 4.479-5.199 5.299v1.06c0 2.04 1.14 3.36 3 3.36 1.538 0 2.56-.809 3.192-1.74V5.981c0-.32.18-.48.44-.48h1.28c.3 0 .42.18.42.48 0 1.2-.08 3.279-.08 4.839 0 1.14.28 1.96 1.3 1.96 1.74 0 2.92-3.34 2.92-6.059v-.74c0-.3.14-.48.4-.48h1.32c.296 0 .436.18.436.46zm-14.63.719c1.46-.52 3.34-1.88 3.34-3.459 0-.68-.38-1.18-1.08-1.18-1.5 0-2.2 2.48-2.26 4.639zM22.112 8.14c-.44-.46-1.32-1.02-2.26-1.02-1.58 0-2.76 1.24-2.76 3.08 0 1.56.88 2.579 2.06 2.579 2.08.001 2.96-2.939 2.96-4.639zm39.415-3.959h-1.32c-.26 0-.42.2-.42.5 0 1.52.06 3.499.06 5.459 0 1.42-.36 2.98-1.78 2.98-.46 0-.86-.3-1.04-.46-.16-.14-.22-.18-.3-.18-.18 0-1 1.08-1 1.32s.84 1.16 2.34 1.16c2.78 0 3.779-2.5 3.779-4.759l.02-5.559c.001-.261-.119-.461-.339-.461zm39.434-.36c.78 0 1.24-.52 1.24-1.24 0-.76-.52-1.16-1.2-1.16-.72 0-1.24.5-1.24 1.18 0 .76.5 1.22 1.2 1.22zm24.381 8.559c0 .08-.04.18-.08.24-.66 1.02-1.94 2-3.899 2-1.47 0-2.827-.665-3.677-1.815-.728.968-1.802 1.815-3.256 1.815-1.54 0-2.84-1.1-2.84-2.72 0-1.14.12-2.12.12-3.22 0-.96-.36-1.56-1.24-1.56-1.86 0-2.92 3.72-2.92 6.059v.72c0 .32-.12.5-.44.5h-1.2c-.32 0-.44-.16-.44-.5v-.239c-.667.569-1.504.959-2.53.959-1.188 0-2.087-.504-2.589-1.414-.748.799-1.794 1.414-3.213 1.414-1.283 0-2.312-.555-2.9-1.518-.738.861-1.759 1.518-3.072 1.518-1.58 0-2.54-.92-2.54-2.2 0-.18.02-.48.02-.58l-.08-.02c-.5 1.3-1.68 2.8-3.56 2.8-1.9 0-2.84-1.28-2.84-3.36V7.67c-.435.126-.939.21-1.498.21-.44 0-.86-.06-1.22-.18.3.64.46 1.36.46 2.14 0 2.38-1.58 4.779-4.779 4.779-1.596 0-2.878-.658-3.668-1.748-.778 1.089-1.916 2.088-3.357 2.088-1.72 0-2.4-1.14-3.2-2.78-.7-1.44-1.2-2.18-1.94-2.5-.2-.12-.32-.2-.32-.38V7.94c0-.18.08-.28.24-.28h.3c1.28 0 2.42-.98 2.42-2.48 0-1.96-1.66-3-4.059-3-3.2 0-5.159 1.92-5.159 3.999 0 1.32.76 2.14 1.5 2.44.08.02.14.06.14.16 0 .22-.46 1.04-.64 1.28-.08.12-.14.16-.22.16-.54 0-2.76-1.04-2.76-4.059 0-3.359 2.96-5.839 7.159-5.839 3.34 0 6.019 1.74 6.019 4.779 0 2-1.48 3.479-3.18 3.579v.08c1.08.24 1.54 1.08 2.26 2.48.62 1.24 1.02 1.88 1.82 1.88.946 0 1.732-.977 2.246-1.89a5.47 5.47 0 0 1-.14-1.23c0-2.68 1.94-4.739 4.779-4.739 1 0 1.88.26 2.58.72.18-.34.5-.62.66-.62.22 0 .8.96 2.18.96.533 0 .994-.111 1.378-.26v-.08c0-.32.16-.48.44-.48h1.26c.3 0 .44.18.44.46 0 1.22-.08 3.499-.08 4.839 0 1.32.36 1.98 1.34 1.98 1.56 0 2.88-3.04 2.88-6.099v-.7c0-.3.14-.48.44-.48h1.26c.3 0 .46.16.46.48 0 .68-.18 3.139-.18 4.979 0 .98.36 1.82 1.24 1.82.833 0 1.55-.811 2.038-1.621-.001-.052-.005-.104-.005-.158V7.24c-.24.02-.46.02-.6.02-.28 0-.4-.1-.4-.36v-.96c0-.2.1-.36.34-.36.18 0 .4 0 .66.02V3.22c0-.3.14-.48.42-.48h1.24c.28 0 .42.18.42.48v2.4c.66-.02 1.5-.04 1.999-.04.16 0 .28.1.28.32v1.02c0 .24-.1.34-.3.34-.5 0-1.3-.02-2-.04v2.88c0 1.66.4 2.68 1.8 2.68 1.051 0 1.831-.764 2.342-1.561V5.961c0-.28.14-.46.42-.46h1.28c.3 0 .44.16.44.48 0 .92-.08 3.699-.08 4.739 0 1.24.44 2.06 1.44 2.06.853 0 1.573-.757 2.07-1.554V6.02c0-.38.14-.52.46-.52h1.18c.3 0 .44.14.44.48 0 .32-.1 1.48-.14 2.02l.08.02c.6-1.42 1.74-2.76 3.639-2.76 1.74 0 2.76 1.06 2.76 2.68 0 .58-.14 2-.14 3.2 0 1.02.38 1.64 1.16 1.64.804 0 1.515-.754 2.023-1.609a5.215 5.215 0 0 1-.148-1.231c0-2.46 1.78-4.679 4.199-4.679 1.66 0 2.859 1.12 2.859 2.62 0 2.24-1.88 3.419-3.959 3.419-.26 0-.64-.02-.82-.04.28.98 1.24 1.66 2.459 1.66 1.18 0 2.08-.68 2.62-1.34.06-.08.1-.12.18-.12.159 0 1.019.7 1.019.92zM77.731 9.8c0-1.6-.86-2.82-2.38-2.82-1.6 0-2.6 1.44-2.6 3 0 1.76 1.06 2.939 2.48 2.939 1.58.001 2.5-1.379 2.5-3.119zm40.972-.28c0 .1.02.2.02.28.18.02.36.02.64.02 1.34 0 2.72-.62 2.72-1.96 0-.64-.44-1.02-1.12-1.02-1.26 0-2.26 1.26-2.26 2.68z"
      />
    </svg>
  );
}

export default Logo;
