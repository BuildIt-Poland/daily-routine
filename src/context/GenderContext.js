import React, { useState, createContext } from 'react';
import PropTypes from 'prop-types';

import { MALE } from '../constants/genders';

const GenderContext = createContext();
const { Provider } = GenderContext;

function GenderProvider({ children }) {
  const handleGenderChange = gender => setGender(gender);

  const [gender, setGender] = useState(MALE);

  return <Provider value={{ gender, handleGenderChange }}>{children}</Provider>;
}

GenderProvider.propTypes = {
  children: PropTypes.node
};

export { GenderContext, GenderProvider };
