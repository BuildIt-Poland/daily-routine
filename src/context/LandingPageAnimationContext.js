import React, { createContext, useState } from 'react';
import PropTypes from 'prop-types';

const LandingPageAnimationContext = createContext();
const { Provider } = LandingPageAnimationContext;

function LandingPageAnimationProvider({ children, history }) {
  const REDIRECT_DELAY = 1000;

  function redirectWithDelay(url) {
    setTimeout(() => {
      history.push(url);
    }, REDIRECT_DELAY);
  }

  function animateAndRedirect(redirectUrl) {
    toggleAnimation(state => ({
      ...state,
      isAnimated: true
    }));
    redirectWithDelay(redirectUrl);
  }

  function stopAnimation() {
    toggleAnimation(state => ({
      ...state,
      isAnimated: false
    }));
  }

  const initialState = {
    isAnimated: false,
    animateAndRedirect,
    stopAnimation
  };

  const [animation, toggleAnimation] = useState(initialState);

  return <Provider value={animation}>{children}</Provider>;
}

LandingPageAnimationProvider.propTypes = {
  children: PropTypes.node,
  history: PropTypes.object
};

export { LandingPageAnimationContext, LandingPageAnimationProvider };
