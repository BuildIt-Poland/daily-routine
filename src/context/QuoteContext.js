import React, { createContext, useState } from 'react';
import PropTypes from 'prop-types';

const QuoteContext = createContext();
const { Provider } = QuoteContext;

function QuoteProvider({ children }) {
  const [quote, setQuote] = useState(null);

  const handleQuoteChange = quote => setQuote(quote);

  return <Provider value={{ quote, handleQuoteChange }}>{children}</Provider>;
}

QuoteProvider.propTypes = {
  children: PropTypes.node
};

export { QuoteContext, QuoteProvider };
