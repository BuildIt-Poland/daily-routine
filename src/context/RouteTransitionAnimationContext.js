import React, { createContext, useState } from 'react';
import PropTypes from 'prop-types';

import { history } from '../types';

const RouteTransitionAnimationContext = createContext();
const { Provider } = RouteTransitionAnimationContext;

function RouteTransitionAnimationProvider({ children, history }) {
  const REDIRECT_DELAY = 250;

  function redirectWithDelay(url) {
    setTimeout(() => {
      history.push(url);
    }, REDIRECT_DELAY);
  }

  function animateAndRedirect(redirectUrl) {
    setIsAnimating(true);
    redirectWithDelay(redirectUrl);
  }

  function stopAnimation() {
    setIsAnimating(false);
  }

  const [isAnimating, setIsAnimating] = useState(false);

  return <Provider value={{ isAnimating, animateAndRedirect, stopAnimation }}>{children}</Provider>;
}

RouteTransitionAnimationProvider.propTypes = {
  children: PropTypes.node,
  history: history.isRequired
};

export { RouteTransitionAnimationContext, RouteTransitionAnimationProvider };
