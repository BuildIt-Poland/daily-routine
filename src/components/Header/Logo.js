import React from 'react';
import { Link as LinkBase } from 'react-router-dom';
import styled from 'styled-components';

import { colorBlack, spacingSmall, borderWidthThin, fontSmall } from '../../styles/designTokens';
import { Logo as LogoIcon } from '../Icons';

const Link = styled(LinkBase)`
  display: flex;
  text-decoration: none;
  align-items: center;
  height: 100%;
  cursor: pointer;
`;

const LogoWrapper = styled.div`
  padding-top: 0.6rem;
  padding-right: ${spacingSmall};
  margin-right: ${spacingSmall};
  border-right: ${borderWidthThin} solid ${colorBlack};
`;

const Tagline = styled.p`
  margin: 0;
  padding: 0;
  color: ${colorBlack};
  font-size: 1.2rem;
  line-height: 1;
  text-transform: uppercase;
`;

function Logo() {
  return (
    <Link to="/">
      <LogoWrapper>
        <LogoIcon />
      </LogoWrapper>
      <Tagline>
        Daily scrum <br />
        absurd answers
      </Tagline>
    </Link>
  );
}

export default Logo;
