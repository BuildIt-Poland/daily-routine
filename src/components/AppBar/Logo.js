import React, { useContext } from 'react';
import { Link as LinkBase } from 'react-router-dom';
import styled from 'styled-components';

import { colorBlack, spacingSmall, borderWidthThin } from '../../styles/designTokens';
import { Logo as LogoIcon } from '../Icons';
import { LandingPageAnimationContext } from '../../context/LandingPageAnimationContext';
import { ROOT_PATH } from '../../constants/routes';
import { location } from './../../types';
const Link = styled(LinkBase)`
  display: flex;
  text-decoration: none;
  align-items: center;
  height: 100%;
  cursor: pointer;
`;

const LogoWrapper = styled.div`
  padding-top: 0.6rem;
  padding-right: ${spacingSmall};
  margin-right: ${spacingSmall};
  border-right: ${borderWidthThin} solid ${colorBlack};
`;

const Tagline = styled.p`
  margin: 0;
  padding: 0;
  color: ${colorBlack};
  font-size: 1.2rem;
  line-height: 1.2;
  text-transform: uppercase;
`;

function Logo({ location }) {
  const { animateAndRedirect } = useContext(LandingPageAnimationContext);

  function handleClick(e) {
    e.preventDefault();
    if (location.pathname !== ROOT_PATH) {
      animateAndRedirect(ROOT_PATH);
    }
  }

  return (
    <Link onClick={handleClick} to={ROOT_PATH} data-testid="logo">
      <LogoWrapper>
        <LogoIcon />
      </LogoWrapper>
      <Tagline>
        Daily scrum <br />
        absurd answers
      </Tagline>
    </Link>
  );
}

Logo.propTypes = {
  location
};

export default Logo;
