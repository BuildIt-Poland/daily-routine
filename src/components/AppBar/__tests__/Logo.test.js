import React from 'react';
import { create } from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Logo from '../Logo';
import { LandingPageAnimationContext } from '../../../context/LandingPageAnimationContext';

jest.mock('../../Icons', () => ({
  Logo: 'LogoIcon'
}));

describe('COMPONENT - AppBar Logo', () => {
  it('renders correctly', () => {
    const animateAndRedirect = () => {};

    const component = create(
      <LandingPageAnimationContext.Provider value={{ animateAndRedirect }}>
        <MemoryRouter initialEntries={['/']}>
          <Logo />
        </MemoryRouter>
      </LandingPageAnimationContext.Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
