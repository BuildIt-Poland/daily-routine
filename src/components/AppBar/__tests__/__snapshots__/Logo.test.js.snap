// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - AppBar Logo renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  cursor: pointer;
}

.c1 {
  padding-top: 0.6rem;
  padding-right: 1rem;
  margin-right: 1rem;
  border-right: 1px solid #222222;
}

.c2 {
  margin: 0;
  padding: 0;
  color: #222222;
  font-size: 1.2rem;
  line-height: 1.2;
  text-transform: uppercase;
}

<a
  className="c0"
  data-testid="logo"
  href="/"
  onClick={[Function]}
>
  <div
    className="c1"
  >
    <LogoIcon />
  </div>
  <p
    className="c2"
  >
    Daily scrum 
    <br />
    absurd answers
  </p>
</a>
`;
